test/connection_switching_test.rb

```diff
@@ -17,6 +17,41 @@ def clear_connection_pool
     require 'models'
   end
 
+  describe "on_primary_db" do
+    after do
+      ActiveRecord::Base.default_shard = nil
+    end
```

Comment:
This is to restore the default_shard after we override it in a test below as to not leak over and affect other tests.